File: trunk/speed/blur_opencv.py
#!/usr/bin/env python

"""
Load, Filter then Show an image with opencv in python
Brian Thorne 2009
"""

from VideoCapturePlayer import VideoCapturePlayer as VCP
from opencv import cv

def gaussianBlur(image, filterSize=43):
    """Blur an image with a particular strength filter.
    Default is 43, 139 gives a very strong blur, but takes a while"""
    
    # Carry out the filter operation
    cv.cvSmooth(image, image, cv.CV_GAUSSIAN, filterSize)
    return image

if __name__ == "__main__":
    title = "Guassian Filtered Output"
    VCP(gaussianBlur, title).main()

File: trunk/speed/blur_opencv.cxx
/*
 *      blur_opencv.cxx
 *      
 *      Copyright 2009 Brian Thorne <brian.thorne@canterbury.ac.nz>
 * 
 *      Compile with:
 * 
 * g++ -Wall `pkg-config --cflags opencv` `pkg-config --libs opencv` -o blur blur_opencv.cxx videoCapturePlayer.cxx
 *  
 */


#include <iostream>
#include "videoCapturePlayer.h"

CvMat * gaussianBlur(CvMat *x)
{
    // Filter with gaussian smoothing
    int filterSize = 43;
    cvSmooth(x, x, CV_GAUSSIAN, filterSize);
    return x;
}

int main( int argc, char** argv )
{
    std::cout << "Starting Gassian Blur C++ demo" << std::endl;
    std::cout << "Press 'q' to exit the program" << std::endl;
    VideoCapturePlayer vcp = VideoCapturePlayer(&gaussianBlur);
    vcp.init(); vcp.main();
    return 0;
}

